src/ServiceInsight.Tests/XamlIconsTests.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                InitWPFResourcesSystem();
+
+                var xamlIcons = GetXamlIconsResourceDictionary();
+
+                results = xamlIcons.Cast<DictionaryEntry>()
+                    .Where(e => e.Value is TValue)
+                    .Select(e => new TestCaseData((string)e.Key).SetName((string)e.Key))
+                    .ToArray();
+
+                wait.Release();
+            });
+            var t = new Thread(ts);
+            t.SetApartmentState(ApartmentState.STA);
+            t.Start();
+            wait.Wait();
+            return results;
+        }
+
+        private static void PrepareForRender(UIElement control)
+        {
+            // From http://stackoverflow.com/a/2596035
+            control.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            control.Arrange(new Rect(control.DesiredSize));
+
+            control.Dispatcher.Invoke(() => { }, DispatcherPriority.Loaded);
+        }
+
+        public static IEnumerable GetControlTemplates => GetTestCases<ControlTemplate>();
+
+        public static IEnumerable GetDrawingImages => GetTestCases<DrawingImage>();
+
+        public static IEnumerable GetGeometries => GetTestCases<Geometry>();
+
+        [Test]
+        [Explicit]
+        [TestCaseSource(typeof(XamlIconsTests), nameof(GetControlTemplates))]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void ControlTemplateTest(string resourceName)
+        {
+            var icon = new ContentControl { Foreground = Brushes.Green };
+            var control = CreateDisplayFrame(resourceName, icon);
+            icon.Template = (ControlTemplate)control.FindResource(resourceName);
+
+            PrepareForRender(control);
+
+            NamerFactory.Clear();
+            using (ApprovalResults.ForScenario(resourceName))
+            {
+                Approvals.Verify(new ImageWriter(f => WpfUtils.ScreenCapture(control, f)));
+            }
+        }
+
+        [Test]
+        [Explicit]
+        [TestCaseSource(typeof(XamlIconsTests), nameof(GetDrawingImages))]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void DrawingImagesTest(string resourceName)
+        {
+            var icon = new Image();
+            var control = CreateDisplayFrame(resourceName, icon);
+            icon.Source = (DrawingImage)control.FindResource(resourceName);
+
+            PrepareForRender(control);
+
+            NamerFactory.Clear();
+            using (ApprovalResults.ForScenario(resourceName))
+            {
+                Approvals.Verify(new ImageWriter(f => WpfUtils.ScreenCapture(control, f)));
+            }
+        }
+
+        [Test]
+        [Explicit]
+        [TestCaseSource(typeof(XamlIconsTests), nameof(GetGeometries))]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void GeometriesTest(string resourceName)
+        {
+            var icon = new Path { Fill = Brushes.Black, Stretch = Stretch.UniformToFill };
```

Comment:
With `Stretch = Stretch.UniformToFill` the image looks like this, with the circle cropped:

![image](https://cloud.githubusercontent.com/assets/677704/20975430/cebbfd88-bc9f-11e6-8593-0ec984a15708.png)

but with `Stretch = Stretch.Uniform`, it looks like this, with the entire circle shown:

![image](https://cloud.githubusercontent.com/assets/677704/20975560/277fddfe-bca0-11e6-87af-0885b8fb1d5b.png)

Would it be better to have the latter so that the full geometry is shown?